### ISSUE TITLE
vmalert: support multiple remote-write targets

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#694]
**Is your feature request related to a problem? Please describe.**
For now `vmalert` may be configured with single remote-write endpoint for rules and alerts persistent. But for some cases
having ability to write into multiple storages is required since it opens new possibilities like writing into storages with different retention, relabeling or downsample settings. 

**Describe the solution you'd like**
Ability to configure multiple remote-write endpoints.

**Describe alternatives you've considered**
 Configure `vmalert` to send write request to `vmagent` and use it as demultiplexer.

**Comment by raags** [comment]
This also allows making vmalert HA. However, writing to vmagent instead has the advantage of buffering, that vmalert wouldn't have.

**Comment by wu0407** [comment]
record rules need multiple remote-write targets like vmagent

**Comment by kovalev94** [comment]
This feature allows making vmalert HA by container orchestrator tools. I think it more correct way to realize HA.
Also vmagent have this feature, so vmalert shouldn't stand out.

**Comment by guinguo** [comment]
That sounds good 😃

**Comment by iamhritikkumar** [comment]
@hagen1778  I have one query ->  When sending write requests from vmalert to vmagent, is it necessary to include a scrape_config job in the vmagent configuration? Or will vmagent automatically scrape those metrics and send them to the specified remoteWrite URLs?

**Comment by hagen1778** [comment]
> When sending write requests from vmalert to vmagent, is it necessary to include a scrape_config job in the vmagent configuration? 

No, scrape config is used only when you want vmagent to [scrape](https://docs.victoriametrics.com/keyConcepts.html#pull-model) something. If you want to [push](https://docs.victoriametrics.com/keyConcepts.html#push-model) data, then no scrape config needed. 

Just run a vmagent and configure it to forward data to destinations you want. And vmalert should be instructed to forward data to this vmagent. That would be it.

**Comment by jiekun** [comment]
Any update for this enhancement?

**Comment by dennisme** [comment]
Hey I anyone working on this? Cool if I snag it?

**Comment by hhstu** [comment]
Multiple remoteewirte is useful for record rules in environments where both vmsingles are used and the ha (ha) condition is maintained.

**Comment by hagen1778** [comment]
> Multiple remoteewirte is useful for record rules in environments where both vmsingles are used and the ha (ha) condition is maintained.

It is recommended to use vmagent for replication or any other data modifications.

vmalert's area responsibility is to execute the query and send the result or notification. Adding multiple remote-writes to it opens a way for additional questions:
1. What to do if one of the destinations is down?
2. If vmalert can write to 2 destinations, then should it execute rules against 2 destinations?
3. Can vmalert send only specific recording rules to a specific destination?
4. Should relabeling happen per specific remote destination?

All these questions are already answered by combination of vmagent and vmauth for writes and reads respectively. With these two components user can build flexible high-available multilayer pipelines, with plenty of functionality on each level. I see no need in porting this functionality from vmagent/vmauth to vmalert.
